### PR TITLE
Add an example using augmentations for Temporal GPs

### PR DESCRIPTION
**Pull request by theogf**
As it was done on the 24/06

### CODE REVIEW

File: examples/augmented_inference.jl [hunk 1]
```diff
@@ -0,0 +1,82 @@
+using AbstractGPs
+using TemporalGPs
+using Distributions
+using StatsFuns
+using LinearAlgebra
+using ProgressMeter
```
Comment (willtebbutt) [inline examples/augmented_inference.jl:6]: Could we maybe remove the ProgressMeter stuff? Just adds an extra dep and it's not totally necessary.

File: examples/augmented_inference.jl [hunk 2]
```diff
@@ -0,0 +1,82 @@
+using AbstractGPs
+using TemporalGPs
+using Distributions
```
Comment (willtebbutt) [inline examples/augmented_inference.jl:3]: Could you add Distributions, StatsFuns and LinearAlgebra to the Project + Manifest if they're not already there?

File: examples/augmented_inference.jl [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+using StatsFuns
+using LinearAlgebra
+using ProgressMeter
+
+# Utilising TemporalGPs.jl to work with AbstractGPs requires minimal modification to the
```
Comment (willtebbutt) [inline examples/augmented_inference.jl:8]: Could you replace this generic blurb with something specific to this script? Probably good to point to your paper, and point out algorithm 1 or something.

File: examples/augmented_inference.jl [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+# Build a GP as per usual, and wrap it inside a TemporalGPs.jl object.
+f_raw = GP(Matern52Kernel());
+f = to_sde(f_raw, SArrayStorage(Float64));
+
+# Specify a collection of inputs. Must be increasing.
+T = 1_000;
+x = RegularSpacing(0.0, 1e-1, T);
+
+# Generate some noisy synthetic data from the GP.
```
Comment (willtebbutt) [inline examples/augmented_inference.jl:29]: This comment could do with a tweak.

File: examples/augmented_inference.jl [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+# Compute the exact posterior marginals at `x_pr`.
+f_post_marginals = marginals(f_post(x_pr));
+m_post_marginals = mean.(f_post_marginals);
+σ_post_marginals = std.(f_post_marginals);
+
+# Generate a few posterior samples. Not fantastically-well optimised at present.
+f_post_samples = [rand(f_post(x_pr)) for _ in 1:5];
+
+# Visualise the posterior. The if block is just to prevent it running in CI.
+# if get(ENV, "TESTING", "FALSE") == "FALSE"
```
Comment (willtebbutt) [inline examples/augmented_inference.jl:72]: Would you mind adding this conditional back in? Keen to have this example running in CI.

**Review by willtebbutt — COMMENTED** [review]
This is broadly fine, but it would be good to tidy it up a bit before merging.

I've left some comments, but could you also add it with the other examples in `test/runtests.jl`?

**Comment by theogf** [comment]
Sure it was just he initial PR to not forget! I'll polish it now

**Comment by willtebbutt** [comment]
Thanks for tidying this up -- looks like there are just a couple of typos or something preventing the examples CI from passing.

**Comment by theogf** [comment]
That's what happens when you edit only on the Github website :smile:

**Comment by willtebbutt** [comment]
I'm just going to wait for the examples CI to finish, then I'll merge.

**Comment by willtebbutt** [comment]
Merging because examples have passed.

**Pull request merged**